analyses/infercnv-consensus-cell-type/template-notebooks/SCPCP000015_explore-infercnv-results.Rmd

```diff
@@ -0,0 +1,418 @@
+    remove = FALSE
+  )
+
+
+# Extract a version of the data frame with just the infercnv metadata, not CNVs themselves
+infercnv_metadata_df <- infercnv_df |>
+  dplyr::select(
+    -starts_with("has_"),
+    -starts_with("proportion_"),
+    -starts_with("top_")
+  )
+
+# Read SCE and save UMAP coordinates
+umap_df <- readr::read_rds(sce_file) |>
+  scuttle::makePerCellDF(use.dimred = "UMAP") |>
+  dplyr::select(
+    barcodes,
+    UMAP1 = UMAP.1,
+    UMAP2 = UMAP.2
+  )
+
+# Define cell types based on ewing module
+celltype_df <- readr::read_tsv(ewings_tsv) |>
+  # Label cells as:
+  # - tumor-immune: Cells labeled as tumor by cell-type-ewings, but immune by cell-type-consensus
+  # - tumor-endo: Cells labeled as tumor by cell-type-ewings, but endothelial by cell-type-consensus
+  # - tumor: Cells labeled as tumor by cell-type-ewings
+  # - their existing ewing_annotation if none of the above
+  dplyr::mutate(
+    celltype = dplyr::case_when(
+      ## tumor-immune
+      consensus_annotation %in% immune_labels &
+        stringr::str_starts(ewing_annotation, "tumor") ~ "tumor-immune",
+      ## tumor-endo
+      consensus_annotation %in% endo_labels &
+        stringr::str_starts(ewing_annotation, "tumor") ~ "tumor-endo",
+      ## tumor
+      stringr::str_starts(ewing_annotation, "tumor") ~ "tumor",
+      ## default
+      .default = ewing_annotation
+    )
+  ) |>
+  dplyr::select(barcodes, celltype)
+
+umap_df <- umap_df |>
+  dplyr::left_join(celltype_df)
+```
+
+
+## InferCNV Heatmap
+
+Below, we show the `inferCNV` heatmap for this library.
+To help contextualize this image, here is the table of known CNVs in Ewing sarcoma used for validation:
+
+```{r}
+validation_df
+```
+
+
+![InferCNV heatmap](`r infercnv_png`)
+
+
+## Distribution of CNVs per cell
+
+This section contains figures showing the total number of CNVs detected per cell.
+
+### CNVs in reference vs query cells
+
+The figures below compare the number of CNVs per cell between reference and query cells.
+Ideally, we'll see a bimodal distribution where reference cells are at/near zero, and there is a peak of higher values which may represent tumor cells.
+In the density plots below, distributions are colored based on whether the cells are in the reference or not (aka, "unknown" cells).
+
+```{r}
+# First, construct data frame of total `has_cnv` counts, with other variables needed for plots
+has_cnv_df <- infercnv_df |>
+  dplyr::select(
+    barcodes,
+    starts_with("has_cnv_")
+  ) |>
+  tidyr::pivot_longer(
+    starts_with("has_cnv_"),
+    names_to = "chr",
+    values_to = "cnv"
+  ) |>
+  # sum all values for each
+  dplyr::group_by(barcodes) |>
+  dplyr::summarize(total_cnv_per_cell = sum(cnv)) |>
+  dplyr::ungroup() |>
+  # bring back metadata
+  dplyr::inner_join(
+    infercnv_metadata_df,
+    by = "barcodes"
+  ) |>
+  # join in SCE information
+  dplyr::left_join(umap_df)
+```
+
+
+```{r fig.width = 10, fig.height = 5}
+ggplot(has_cnv_df) +
```

Comment:
It would be helpful to add a plot that is in between this one and the next one where we have: 
- reference cells
- any cells labeled as tumor by Ewings
- any cells that are not in the reference and are not tumor by the Ewings module 

Generally, I want to see if the cells we label as tumor are in fact the cells with higher CNVs. I also want to know what the in between group looks like. If we aren't labeling them as tumor, but they aren't included in the reference are they more similar to the reference or more similar to tumor cells based on their CNV profile. I would also use these groups throughout to simplify some of the plots.

Reply:
> I would also use these groups throughout to simplify some of the plots.

Yeah, this is a much better grouping. I'm going to more broadly apply this.